File: lib/DecryptIdfa.php
<?php
require_once( __DIR__ . DIRECTORY_SEPARATOR . "Decrypt.php" );
class DecryptIdfa extends Decrypt{
   function __construct($encryption_encoded_key, $integrity_encoded_key){
      $iv_length = 16;
      $is_length = 4; 
      $byte_length = 16;
      parent::__construct( $encryption_encoded_key, $integrity_encoded_key, $iv_length, $is_length, $byte_length );
   }
   function decrpyt($long_ciphertext){
      $res = parent::run( $long_ciphertext );
       if(isset( $res['error']) ){
           return $res;
       }else{
           return array(
              'idfa'       => $res["plaintext"],
              'datetime'   => $res["datetime"],
           );
       }
   }
}
?>

File: python/lib/DecryptIdfa.py
import sys,os
from Decryption import Decryption
class DecryptIdfa(Decryption):
    def __init__( self, encryption_encoded_key, integrity_encoded_key):
        iv_length = 16;
        is_length = 4; 
        byte_length = 16;
        self.decrypt = Decryption.__init__( self, encryption_encoded_key, integrity_encoded_key, iv_length, is_length, byte_length );
    def decryption( self, long_ciphertext ):
        res = Decryption.run( self, long_ciphertext );
        if "error" in res:
            return res;
        else:
            return {
                'idfa'     : res["plaintext"],
                'datetime' : res["datetime"],
            };
